### ISSUE TITLE
Consider checking $function_to_add signature against $accepted_args

### ISSUE THREAD

**Issue by ssnepenthe** [humanmade/psalm-plugin-wordpress#17]
This is already implicitly checked since you are using $accepted_args to build up the list of params for $function_to_add, so it's understandable if there is no interest in this.

Still - it could be nice for Psalm to explicitly tell us that there is a mismatch between the two.

For example, given a call to add_filter where I forget to set $accepted_args like:

```php
add_filter( 'some_filter', function( One $one, Two $two ) { return $one; } );
```

Psalm might show me something along the lines of:

```
ERROR: InvalidArgument - test.php:29:5 - Argument 2 of add_filter expects callable(One):(One), pure-Closure(One, Two):(One) provided (see https://psalm.dev/004)
```

As opposed to something like:

```
ERROR: TooFewArguments - test.php:29:5 - add_filter $accepted_args is 1 but some_filter callback accepts 2 arguments
```

The message could definitely use some work... Also TooFewArguments and TooManyArguments seemed to be the most sensible built-in issues I could find but a custom issue might make more sense.

**Comment by joehoyle** [comment]
Hmm yes good point, I'm not sure if the hooks are there in Psalm to be able to detect that and output a custom error. The way this works right now is the _type_ of the argument is set, but it's Psalm internally that finds the type mismatch, and then creates that error.

I'd rather not have to implement the type comparison logic myself, but need a way to generate nicer errors.

**Comment by ssnepenthe** [comment]
Makes sense... Looking at it a bit closer now I see that this is probably much more work than I originally assumed.

I thought (or hoped) that in Plugin::getFunctionParams() the callable at $call_args[1] would be some sort of callable representation that would have already done the heavy lifting for you. But now that I look a bit closer I see these are mostly going to be simple string nodes, array nodes, etc.

**Comment by kkmuffme** [comment]
This PR https://github.com/humanmade/psalm-plugin-wordpress/pull/31 adds exactly this feature (and some more)

**Comment by kkmuffme** [comment]
This should work now in v3 https://github.com/psalm/psalm-plugin-wordpress/releases/tag/3.0.0
